ci/test-sriov-secondary-network-aws.sh

```diff
@@ -0,0 +1,582 @@
+
+_usage="Usage: $0 [--aws-access-key <AccessKey>] [--aws-secret-key <SecretKey>] \
+                  [--aws-security-group-id <SecurityGroupID>] [--aws-subnet-id <SubnetID>] \
+                  [--aws-ec2-ssh-key-name <SSHKeyName>]
+                  [--aws-service-user-role-arn <ServiceUserRoleARN>]  [--aws-service-user <ServiceUserName>] \
+                  [--aws-region <Region>] [--k8s-version <ClusterVersion>]
+
+Setup a Kubernetes cluster and test SR-IOV secondary network in AWS.
+
+        --aws-access-key              AWS Acess Key for logging in to awscli.
+        --aws-secret-key              AWS Secret Key for logging in to awscli.
+        --aws-security-group-id       Security group for the ec2 instance in the Kubernetes cluster.
+        --aws-subnet-id               The subnet in which the ec2 instance network interface is located.
+        --aws-ec2-ssh-key-name        The key name to be used for ssh access to ec2 instances.
+        --aws-service-user-role-arn   AWS Service User Role ARN for logging in to awscli.
+        --aws-service-user            AWS Service User Name for logging in to awscli.
+        --aws-region                  The AWS region where the cluster will be initiated. Defaults to $REGION.
+        --setup-only                  Only perform setting up the cluster and run test.
+        --cleanup-only                Only perform cleaning up the cluster.
+        --k8s-version                 The K8s cluster version. Defaults to $K8S_VERSION."
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --aws-access-key)
+    AWS_ACCESS_KEY="$2"
+    shift 2
+    ;;
+    --aws-secret-key)
+    AWS_SECRET_KEY="$2"
+    shift 2
+    ;;
+    --aws-security-group-id)
+    AWS_SECURITY_GROUP="$2"
+    shift 2
+    ;;
+    --aws-subnet-id)
+    AWS_SUBNET_ID="$2"
+    shift 2
+    ;;
+    --aws-ec2-ssh-key-name)
+    AWS_EC2_SSH_KEY_NAME="$2"
+    shift 2
+    ;;
+    --aws-service-user-role-arn)
+    AWS_SERVICE_USER_ROLE_ARN="$2"
+    shift 2
+    ;;
+    --aws-service-user)
+    AWS_SERVICE_USER_NAME="$2"
+    shift 2
+    ;;
+    --aws-region)
+    REGION="$2"
+    shift 2
+    ;;
+    --k8s-version)
+    K8S_VERSION="$2"
+    shift 2
+    ;;
+    --setup-only)
+    RUN_SETUP_ONLY=true
+    RUN_ALL=false
+    shift
+    ;;
+    --cleanup-only)
+    RUN_CLEANUP_ONLY=true
+    RUN_ALL=false
+    shift
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+mkdir -p ~/.aws
+cat > ~/.aws/config <<EOF
+[default]
+region = $REGION
+role_arn = $AWS_SERVICE_USER_ROLE_ARN
+source_profile = $AWS_SERVICE_USER_NAME
+output = json
+EOF
+cat > ~/.aws/credentials <<EOF
+[$AWS_SERVICE_USER_NAME]
+aws_access_key_id = $AWS_ACCESS_KEY
+aws_secret_access_key = $AWS_SECRET_KEY
+EOF
```

Comment:
I'm surprised we need that, isn't it ok to just export the correct environment variables (as opposed to writing these files)? IMO it would be a bit better if we could avoid writing the credentials to the Jenkins test machine. It would also make it simpler to run this script locally from a dev machine.

Reply:
How does this differ from the project's existing implementation? For example `test-conformance-eks.sh` https://github.com/antrea-io/antrea/blob/main/ci/test-conformance-eks.sh#L191

Reply:
Good point, I didn't realize that we were doing it elsewhere as well. We can keep things consistent.